src/main/services/PromptsManager.ts

```diff
@@ -0,0 +1,102 @@
+import { readFileSync } from "fs";
+import { join } from "path";
+import type { PromptTemplate } from "@shared/types";
+
+export class PromptsManager {
+  private static instance: PromptsManager;
+  private promptsCache = new Map<string, PromptTemplate>();
+  private readonly promptsDir: string;
+
+  private constructor() {
+    this.promptsDir = join(__dirname, "../prompts");
+  }
+
+  public static getInstance(): PromptsManager {
+    if (!PromptsManager.instance) {
+      PromptsManager.instance = new PromptsManager();
+    }
+    return PromptsManager.instance;
+  }
+
+  public getPrompt(promptName: string): PromptTemplate {
+    if (this.promptsCache.has(promptName)) {
+      return this.promptsCache.get(promptName)!;
+    }
+
+    const prompt = this.loadPrompt(promptName);
+    this.promptsCache.set(promptName, prompt);
+    return prompt;
+  }
+
+  private loadPrompt(promptName: string): PromptTemplate {
+    try {
+      const filePath = join(this.promptsDir, `${promptName}.md`);
+      const content = readFileSync(filePath, "utf-8");
+
+      const systemPromptMatch = content.match(
+        /## System Prompt\n([\s\S]*?)(?=\n## |$)/,
+      );
+      const userPromptMatch = content.match(
+        /## User Prompt Template\n([\s\S]*?)(?=\n## |$)/,
+      );
+
+      if (!systemPromptMatch || !userPromptMatch) {
+        throw new Error(`Invalid prompt format in ${promptName}.md`);
+      }
+
+      return {
+        systemPrompt: systemPromptMatch[1].trim(),
+        userPromptTemplate: userPromptMatch[1].trim(),
+      };
+    } catch (error) {
+      console.error(`Failed to load prompt ${promptName}:`, error);
+      throw error;
+    }
+  }
+
+  public renderPrompt(
+    template: string,
+    variables: Record<string, string>,
+  ): string {
+    let rendered = template;
+    for (const [key, value] of Object.entries(variables)) {
+      const placeholder = new RegExp(`{{${key}}}`, "g");
+      rendered = rendered.replace(placeholder, value);
+    }
+    return rendered;
+  }
+
+  public getPhraseGenerationPrompt(
```

Comment:
I'd like the PromptManager to have a generic getPrompt that takes in the prompt name and variables to interpolate. 
By having one method for each prompt, we are polluting this class and not adhering to the single responsibility principle